src/app/model/ChessEngine.ts

```diff
@@ -119,4 +136,31 @@ export class ChessEngine implements ChessModel {
   public possibleMoves(position: Square): Square[] {
     return this.pieceMovesNotCausingAllyKingCheckmate(position);
   }
+
+  private kingWasChecked(chosenPiece: Piece): KingWasChecked | undefined {
+    const kingsSide = chosenPiece.side === Side.WHITE ? Side.BLACK : Side.WHITE;
```

Comment:
changeTurn w sumie powinno się nazywać inaczej, bo nie zmienia tury. W stylu `anotherSide` - wtedy mozna to tutaj wykorzsytac.